src/game/common/rts/energy.cpp

```diff
@@ -0,0 +1,190 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "energy.h"
+#include "audiomanager.h"
+#include "gamelogic.h"
+#include "ini.h"
+#include "object.h"
+#include "player.h"
+#include "playerlist.h"
+
+Energy::Energy() : m_energyProduction(0), m_energyConsumption(0), m_frame(0), m_player(nullptr) {}
+
+void Energy::Init(Player *player)
+{
+    m_energyProduction = 0;
+    m_energyConsumption = 0;
+    m_player = player;
```

Comment:
`m_frame` is not set to 0 here? Hard to tell if it should.

Reply:
isn't no, yeah hard to tell